mono/metadata/threads.c

```diff
@@ -5608,6 +5619,39 @@ mono_threads_add_joinable_runtime_thread (gpointer thread_info)
 	}
 }
 
+static void
+threads_add_pending_native_thread_join_call_nolock (gpointer tid)
+{
+	if (!pending_native_thread_join_calls)
+		pending_native_thread_join_calls = g_hash_table_new (NULL, NULL);
+
+	gpointer orig_key;
+	gpointer value;
+
+	if (!g_hash_table_lookup_extended (pending_native_thread_join_calls, tid, &orig_key, &value))
```

Comment:
Why is this using g_hash_table_lookup_extended when it's not doing anything with orig_key and value?

Reply:
g_hash_table_lookup_extended handles cases when the table includes a value that is 0 or NULL (since it returns a gboolean instead of the value). Using regular g_hash_table_lookup doesn't distinct between a value of 0 or NULL and not existing at all in the table. The same pattern is used in several places in threads.c to handle cases where tid could be 0.